File: backend/B2BApplication/DTO/CatalogueKeyPair2.cs
namespace B2BApplication.DTO
{
    public class CatalogueKeyPair2
    {
        public int id { get; set; }
        public string name { get; set; }
        //public string? parent { get; set; }

        //public int niveau { get; set; }
        public int nbArticle { get; set; }
        public List<CatalogueKeyPair2> value { get; set; } = null;

        public bool isExpendable { get; set; } = false;
    }
}

File: frontend/src/app/models/CatalogueKeyPair2.ts
export class CatalogueKeyPair2{
    id:number
    value:CatalogueKeyPair2[]
    name:string
    parentId:number
    niveau:number
    isExpanded=false
    constructor( id:number,value:CatalogueKeyPair2[],name:string,parentId:number,niveau:number){
        this.id=id
        this.value=value
        this.name=name
        this.parentId=parentId
        this.niveau=niveau
    }
}